apps/cursor/usage_monitor.py

```
@@ -143,39 +181,348 @@ def unique_id(self) -> str:
+        """Require a persisted account and token to identify the same subject."""
+        if claims.get("sub") != account.subject:
+            raise CursorAuthenticationError(
+                "Persisted account subject does not match its token",
+            )
+
+    def _validate_legacy_owner(self, account: CursorAccount) -> None:
+        """Require the migrated account to match the configured legacy owner."""
+        if account.email != self.legacy_account_email:
+            raise CursorAuthenticationError(
+                "Stored legacy token conflicts with its configured owner",
+            )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)